src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -280,6 +285,119 @@ public void shutdown(ExecutorDriver d) {
     log.info("Executor asked to shutdown");
   }
 
+  protected void runHealthChecks(ExecutorDriver driver, Task task) {
+    log.info("Performing health check for task: " + task.taskInfo.getTaskId().getValue());
+    
+    String nodeName = null;
+    String healthCheckCmd = "netstat -plnat | grep ";
+    String taskIdStr = task.taskInfo.getTaskId().getValue();
+      
+    if (taskIdStr.contains(HDFSConstants.DATA_NODE_ID)) {
+      nodeName = HDFSConstants.DATA_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.DATA_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.JOURNAL_NODE_ID)) {
+      nodeName = HDFSConstants.JOURNAL_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.JOURNAL_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.ZKFC_NODE_ID)) {
+      nodeName = HDFSConstants.ZKFC_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.ZKFC_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.NAME_NODE_ID)) {
+      nodeName = HDFSConstants.NAME_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.NAME_NODE_PORT;
+    }
+      
+    try {
+      boolean nodeRegistered = false;
+      if (nodeName != null) {
+        nodeRegistered = runHealthCheckProcess(healthCheckCmd, nodeName);
```

Comment:
Why assign a boolean var just to check it once? Can you just call the healthcheck inside of the if statement?